ivy.el

```diff
@@ -3975,17 +3975,29 @@ characters (previous if ARG is negative)."
   "Overlay used to highlight yanked word.")
 
 (defvar ivy--pulse-timer nil
-  "Timer used to dispose of ivy--pulse-overlay.")
+  "Timer used to dispose of `ivy--pulse-overlay'.")
+
+(defvar ivy-pulse-delay 0.5
+  "Number of seconds to display `ivy-yanked-word' highlight.")
 
 (defun ivy--pulse-region (begin end)
   (if ivy--pulse-overlay
-      (move-overlay ivy--pulse-overlay begin end (current-buffer))
-    (setq ivy--pulse-overlay (make-overlay begin end)))
+      (move-overlay ivy--pulse-overlay
+                    (min begin (overlay-start ivy--pulse-overlay))
+                    (max end (overlay-end ivy--pulse-overlay)))
+    (setq ivy--pulse-overlay (make-overlay begin end))
```

Comment:
In principle, this may highlight the wrong region if the user manages to move the point (say, to another line) between calls to `ivy--pulse-region` and before the timer runs out of time.

Of course in practical terms, it is unlikely to happen.

Reply:
Good point. I'll either revert my change or protect against such a scenario in my next PR on this, unless someone beats me to it.